src/workerd/jsg/modules.c++

```diff
@@ -27,15 +27,20 @@ v8::MaybeLocal<v8::Module> resolveCallback(v8::Local<v8::Context> context,
     auto registry = getModulesForResolveCallback(isolate);
     KJ_REQUIRE(registry != nullptr, "didn't expect resolveCallback() now");
 
-    auto& referrerPath = KJ_ASSERT_NONNULL(registry->resolvePath(referrer),
+    auto ref = KJ_ASSERT_NONNULL(registry->resolve(js, referrer),
         "referrer passed to resolveCallback isn't in modules table");
 
-    kj::Path targetPath = referrerPath.parent().eval(kj::str(specifier));
+    kj::Path targetPath = ref.specifier.parent().eval(kj::str(specifier));
 
-    result = JSG_REQUIRE_NONNULL(registry->resolve(js, targetPath), Error,
+    // If the referrer module is a built-in, it is only permitted to resolve
+    // internal modules. If the worker bundle provided an override for a builtin,
+    // then internalOnly will be false.
+    bool internalOnly = ref.type == ModuleRegistry::Type::BUILTIN;
```

Comment:
Seeing this bool pattern all over the place, maybe make an inline func to reduce the likelihood of a mistake being made since it's such a common pattern.

Reply:
There's currently only two places we make this determination and it's done slightly differently in each case. I think for now I'll keep it as is. We can refactor later if necessary.